### ISSUE TITLE
Branching dialogue does not work

### ISSUE THREAD

**Issue by fleskesvor** [godotengine/escoria#86]
As mentioned in #83, branching dialogue is currently broken. The `ResourcePreloader` in `ui/dd_player.tscn` is always empty when accessed from code, even though scenes have been added from the editor.

**Comment by lavaduderDev** [comment]
The strange thing is, it seems like the dialog_player.gd references the same material 
`type = type + ProjectSettings.get_setting("escoria/platform/dialog_type_suffix")
	var inst = get_resource(type).instance()`
but this works for some reason.

**Comment by fleskesvor** [comment]
They have separate `ResourcePreloaders` though, and they're also attached to different scenes. `ui/dialog_player.tscn`, which works, belong to `globals/game.tscn`, while `ui/dd_player.tscn`, which doesn't work, belong to `globals/main.tscn`.

I think the solution might be to use `call_deferred()` to load the scene, because possibly the game is in a state where the `ResourcePreloader` isn't ready yet when dd_player is being loaded. That would be similar to the case of the bg_music scene that I fixed yesterday at least.

**Comment by fleskesvor** [comment]
What's strange is that the ResourcePreloader is not empty when checked in the `_init()` function, which I thought was only called when a new object was created, and not on instanced scenes. Therefore, copying its contents to a dictionary sort of works, but is an ugly hack. Knowing that might possibly help us understand why this is happening though.

I've made a PR (#87) using this workaround, in case anyone wants to hack away at it. It is not meant to be merged though.

**Comment by StraToN** [comment]
As explained on IRC, I found the cause of this problem:

The `ui/dd_player.tscn` scene is basically a ResourcePreloader node with a `default` resource set in it.
However, its **instance** in `globals/main.tscn` does not have any resource set anymore (it must have been removed at some point). This is what caused the ResourcePreloader to effectively contain a resource in the `_init()` function, and no more right before `_enter_tree()`, which is normal because the effective content of the instance is set at this moment.

Thus, it is easy to fix it: simply remove the instance from `globals/main.tscn` and re-create it.
